reverse.js

```diff
@@ -194,9 +195,16 @@
       }
       else if(lop === "s") {
         for (a = 0; a < alen; a += 4) {
-          // reflect previous cx2/cy2 over x/y
-          cx = x + (x-cx2);
-          cy = y + (y-cy2);
+          if (["s", "c"].includes(preop)) {
+            // reflect previous cx2/cy2 over x/y
+            cx = x + (x-cx2);
+            cy = y + (y-cy2);
+          }
+          else {
+            cx = x;
+            cy = y;
+          }
+
```

Comment:
small tweak, but note that in both cases we assign x/y:
```suggestion
          cx = x;
          cy = y;
          if (["s", "c"].includes(prevop)) {
            cx += (x-cx2);
            cy += (y-cy2);
          }
```